File: Desktop/SQL-Auto_Grader/sql-auto-grader-lite-app/src/components/services/UserSession.js
class UserSession {
  constructor() {
    this._user = null;
  }

  set(data) {
    this._user = data; // { uid, email, fullName, role, createdAt }
  }

  get() {
    return this._user;
  }

  clear() {
    this._user = null;
  }

  get uid()      { return this._user?.uid; }
  get email()    { return this._user?.email; }
  get fullName() { return this._user?.fullName; }
  get role()     { return this._user?.role; }
  get createdAt(){ return this._user?.createdAt; }
}

const userSession = new UserSession();
export default userSession;

File: lib/services/user_session.dart
import '../models/user_model.dart';

class UserSession {
  static UserModel? _user;

  static void set(UserModel? user) => _user = user;
  static UserModel? get() => _user;
  static void clear() => _user = null;

  static String? get uid => _user?.uid;
  static String? get email => _user?.email;
  static String? get fullName => _user?.fullName;
  static String? get role => _user?.role;
}
